### DISCUSSION TITLE
Why is vercel/next prepending vercel.app url to OG meta tags instead of custom domain?

### DISCUSSION THREAD

**Discussion by Levelleor** [vercel/community#4447] [Help]
### Summary

I have this code:
```
		images: [
			{
				url: "/og.jpg",
				width: 1920,
				height: 1080,
			},
		],
```
While this is what's generated on deployment:
```<meta name="twitter:image" content="https://...-levelleor.vercel.app/og.jpg">```

I dont' understand where this url is coming from. I have a custom domain attached, why is vercel.app url attached instead of my custom domain? Makes no sense, I don't want any traffic to ever lead to vercel.app on my site, is there a way to make sure vercel is using my own custom domain everywhere it would use its vercel.app url? 

### Steps to Reproduce

Just deploy a site with images being relative in the metadata and vercel will prepend vercel.app urls to them instead of your custom domain.

**Comment by z4nr34l** [comment]
Next.js is always prerendering pages using demployment URLs which are not custom domains.
Most often solution is to hardcode final absolute url with custom domain.

Currently there is no setting that will let us override build deployment URL to custom domain.

**Reply by Levelleor** [reply]
Got it. I've been trying to find a universal workaround and use at least `metadataBase` for it, but it's just being ignored all the time and vercel keeps using the default VERCEL_URL for some reason. The warning about unset metadataBase is gone but the value is ignored. I got tired of all of this and hardcoded the value as you have mentioned, unless there's some case where url is often-changing this is fine.

**Reply by z4nr34l** [reply]
You can always make part of that URL dynamic, for example a sample from my portfolio:
```tsx
// File: page.tsx
// Method: generateMetadata

    const ogUrl = new URL(`${siteConfig.url}/api/og?title=${post.title}`)

    return {
        // [...]
        openGraph: {
            // [...]
            images: [
                {
                    url: ogUrl.toString(),
                    width: 1920,
                    height: 1080,
                    alt: post.title,
                },
            ],
        },
        twitter: {
            // [...]
            images: [ogUrl.toString()],
        },
    }
```

And inside siteConfig I've made prop url which I can set as I want, even from env vars :D

If that helped consider marking comment as answered to help others find a solution faster.

**Reply by Levelleor** [reply]
Where is it coming from? Do you always set localhost/domain url through env vars?
